wallet-new/src/Cardano/Wallet/API/Response.hs

```diff
@@ -174,18 +180,38 @@ data ValidJSON deriving Typeable
 
 instance FromJSON a => MimeUnrender ValidJSON a where
     mimeUnrender _ bs = case eitherDecode bs of
-        Left err -> Left $ decodeUtf8 $ encodePretty (jsonValidationFailed err)
+        Left err -> Left $ decodeUtf8 $ encodePretty (JSONValidationFailed $ toText err)
         Right v  -> return v
-      where
-        -- NOTE Cheating a bit with type params here, ideally, we would like
-        -- types we can render ToJSON, though we only use the JSONValidationFailed
-        -- which doesn't rely on the type-parameters.
-        jsonValidationFailed :: String -> WalletError () () ()
-        jsonValidationFailed =
-            JSONValidationFailed . toText
 
 instance Accept ValidJSON where
     contentType _ = contentType (Proxy @ JSON)
 
 instance ToJSON a => MimeRender ValidJSON a where
     mimeRender _ = mimeRender (Proxy @ JSON)
+
+
+--
+-- Error from parsing / validating JSON inputs
+--
+
```

Comment:
@KtorZ  I see you have made `MigrationFailed` an instance of `Expcetion` , but JSONValidationError` doesn't have the same instance. Is there a reason for the separation?

Reply:
Hmmm. No. That's an oversight :+1: